packages/sitecore-jss-react/src/components/File.tsx

```diff
@@ -31,6 +36,11 @@ export const File: React.FC<FileProps> = ({ field, children, ...otherProps }) =>
     return null;
   }
 
+  // when metadata is present, render it to be used for chrome hydration
+  if (metadata) {
```

Comment:
More of a general implementation question: we add this condition in every component, and would need to not forget adding it to future components we implement. 
Would it be better to create a new base class (something like MetadataComponent or HOC?) and make all our existing classes inherit it instead, put the metadata logic in there? This would be a much lower maintenance approach

Reply:
yeap, reworking that now with a HOC :)